components/Block.vue

```diff
@@ -75,66 +74,57 @@ export default {
     },
     handleDragStart(e) {
       const { clientX, clientY, offsetX, offsetY } = e
-      const { offsetHeight, offsetWidth } = e.target.offsetParent.parentElement
+      const {
+        height,
+        width
+      } = e.target.parentNode.parentNode.getBoundingClientRect()
 
-      this.ghostDraggingStyle = {
-        height: `${offsetHeight}px`,
-        width: `${offsetWidth}px`,
+      this.ghostStyle = {
+        height: `${height}px`,
+        width: `${width}px`,
         top: `${clientY}px`,
         left: `${clientX}px`,
         transform: `translate(${-Math.abs(offsetX)}px,${-Math.abs(offsetY)}px)`
       }
 
-      this.dragStartCoords.x = clientX
-      this.dragStartCoords.y = clientY
       this.isDragging = true
     },
     handleDrag(e) {
-      e.preventDefault()
       const { clientX, clientY } = e
+      const { x, y } = e.target.parentNode.parentNode.getBoundingClientRect()
```

Comment:
There's definitely duplication here that could probably be extracted to some computed property or something.